06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * InvertedIndex class
+ * @class
+ */
+class InvertedIndex {
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.indexed = {};
+  }
+
+  /**
+   * Get the index for a particular file
+   * @function
+   * @param {String} fileName
+   * @return {Object} index object
+   */
+  getIndex(fileName) {
+    return this.indexed[fileName];
+  }
+
+  /**
+   * Read files using FileReader
+   * @function
+   * @param {String} file
+   * @return {Promise} validateFile response
+   */
+  static readFile(file) {
+    return new Promise((resolve, reject) => {
+      try {
+        const reader = new FileReader();
+        reader.onload = () => {
+          const fileToValidate = JSON.parse(reader.result);
+          const response = InvertedIndex.validateFile(fileToValidate);
+          resolve(response);
+        };
+        reader.readAsText(file);
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  /**
+   * Validate File
+   * @function
+   * @param {string} fileToValidate content of the file uploaded
+   * @return {Object} result.success should return true or false
+   */
+  static validateFile(fileToValidate) {
```

Comment:
Nice work here!

I see you're looping through all the files to validate each of the files. I however can see a logical error here, if you have three documents in a file and the first two are invalid with the last file valid it will still return true for that file.

Is the end goal to test that at least one document is valid or all documents in a file are valid?